pkgs/development/python-modules/openai-triton/default.nix

```diff
@@ -0,0 +1,242 @@
+{ lib
+, buildPythonPackage
+, python
+, fetchpatch
+, fetchFromGitHub
+, addOpenGLRunpath
+, cmake
+, cudaPackages ? { }
+, llvmPackages
+, pybind11
+, gtest
+, zlib
+, ncurses
+, libxml2
+, lit
+, filelock
+, torchWithRocm
+, pytest
+, pytestCheckHook
+, pythonRelaxDepsHook
+, pkgsTargetTarget
+}:
+
+let
+  pname = "triton";
+  version = "2.0.0";
+
+  inherit (cudaPackages) cuda_cudart backendStdenv;
+  ptxas = "${pkgsTargetTarget.cudaPackages.cuda_nvcc}/bin/ptxas";
```

Comment:
what is `pkgsTargetTarget` and why do we pull `cuda_nvcc` out of it instead of simply `cudaPackages`?

Reply:
It's for cross compilation.

The short story is that the ptxas binary we want is not the one for our current build machine, but one that can run on the platform we want to deploy triton on. That's the first "Target" in "pkgsTargetTarget". The second one means is that this ptxas is also supposed to compile for our Target

Reply:
(normally we'd just put in dependencies buildInputs/nativeBuildInputs and nixpkgs magic would work out which package set to take the actual derivation from)

Reply:
(obviously, our situation isn't normal)

Reply:
ahh i see! this is subtle enough i think a comment could be helpful for others

Reply:
Added the comment